crates/biome_graphql_analyze/src/lint/nursery/no_anonymous_operations.rs

```diff
@@ -0,0 +1,120 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_graphql_factory::make;
+use biome_graphql_syntax::GraphqlOperationDefinition;
+use biome_rowan::{AstNode, AstNodeExt, BatchMutationExt};
+use biome_string_case::Case;
+
+use crate::GraphqlRuleAction;
+
+declare_lint_rule! {
+    /// Require specifying name for GraphQL operations.
+    ///
+    /// This is useful since most GraphQL client libraries are using the operation name for caching purposes..
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```graphql,expect_diagnostic
+    /// query {}
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```graphql
+    /// query Human {
+    ///   name
+    /// }
+    /// ```
+    ///
+    pub NoAnonymousOperations {
+        version: "next",
+        name: "noAnonymousOperations",
+        language: "graphql",
+        sources: &[RuleSource::EslintGraphql("no-anonymous-operations")],
+        source_kind: RuleSourceKind::SameLogic,
+        recommended: true,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+impl Rule for NoAnonymousOperations {
+    type Query = Ast<GraphqlOperationDefinition>;
+    type State = NoAnonymousOperationsState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let node = ctx.query();
+        let operation_type = node.ty().ok()?.text();
+        if node.name().is_some() {
+            None
+        } else {
+            Some(NoAnonymousOperationsState {
+                operation_type: operation_type.clone(),
+                suggested_name: get_suggested_name(node, operation_type),
+            })
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Anonymous GraphQL operations are forbidden. Make sure to name your " {state.operation_type}"!"
+                },
+            )
+            .note(markup! {
+                "Rename this "{state.operation_type}" to "{state.suggested_name}"."
```

Comment:
Let's follow our rule pillars https://biomejs.dev/linter/#rule-pillars

This means that the .note must become a message that explains the error. The core action is the solution for fixing the error.